### PR TITLE
mapnik: use `scons` to build instead of bundled copy

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by cho-m** [comment]
Saw some weird build issues locally.
- Sometimes `scons` command would just not work and then work on re-run. 
- `sqlite` was causing some issues for me.

Will need to check on CI how it behaves.

**Comment by chenrui333** [comment]
> * `sqlite` was causing some issues for me.

are you using linux or osx?

**Comment by cho-m** [comment]
> > * `sqlite` was causing some issues for me.
> 
> are you using linux or osx?

Was testing on macOS Ventura on my Rosetta copy of Homebrew.

Ended up hitting https://github.com/mapnik/mapnik/issues/4051. Not sure if it is due to having Xcode app installed.

**Comment by chenrui333** [comment]
It builds/tests fine on my local (ventura arm)

**Comment by cho-m** [comment]
ARM Big Sur hit the issue I saw. Not sure reason, but may just add `sqlite` dependency.
```
In file included from plugins/input/sqlite/sqlite_datasource.hpp:46:
  plugins/input/sqlite/sqlite_connection.hpp:171:9: error: use of undeclared identifier 'sqlite3_enable_load_extension'
          sqlite3_enable_load_extension(db_, 1);
          ^
  plugins/input/sqlite/sqlite_connection.hpp:172:22: error: use of undeclared identifier 'sqlite3_load_extension'
          int result = sqlite3_load_extension(db_, ext_path.c_str(), 0 , 0);
                       ^
```

File: Formula/mapnik.rb [hunk 1]
```diff
@@ -47,55 +48,50 @@ class Mapnik < Formula
...
+    icu = Formula["icu4c"]
+    jpeg = Formula["jpeg-turbo"]
+    libpng = Formula["libpng"]
+    libtiff = Formula["libtiff"]
+    proj = Formula["proj"]
+    webp = Formula["webp"]
 
     args = %W[
       CC=#{ENV.cc}
       CXX=#{ENV.cxx}
```
Comment (cho-m) [inline Formula/mapnik.rb:62]: Ah. Issue is directly depending on `scons` causes `brew` to use `--env=std`. May not be worth the hassle to do this and instead just wait for next release with `cmake` support.

**Review by SMillerDev — APPROVED** [review]

**Review by branchv — APPROVED** [review]
LGTM, @cho-m do you remember why this was still a draft

**Comment by cho-m** [comment]
I think I wasn't sure on effort to get `scons` to work in brew environment and was considering waiting for new release to switch to `cmake`.

However, it looks like new release is still not out so we could just use `scons` for now.

**Review by p-linnane — APPROVED** [review]

**Pull request merged**